src/kernel/filesystem/fat32.zig

```diff
@@ -1247,6 +1308,7 @@ pub fn Fat32FS(comptime StreamType: type) type {
             self.opened_files.deinit();
             self.allocator.destroy(self.root_node.node);
             self.allocator.destroy(self.fs);
+            self.allocator.destroy(self);
```

Comment:
It would be good to specify that this function destroys the pointer as well. Alternatively you could rename `init` to `create` and `deinit` to `destroy` to follow the convention from other structs.

I think option two is best.

Reply:
I think it should still be `init` but have added to the doc comment.

Reply:
How come you kept it as `init`?

Reply:
We haven't really been using `create` for initialising structs (apart from task). So would have to change all the inits to creates. Also, I don't really know what the semantic difference is, aren't they basically the same meaning anyway?

Reply:
I thought the difference between `create` and `init` was that the former allocates on the heap and the latter allocates on the stack. We wouldn't have to change all `init`s to `create`s because most allocate on the stack rather than heap (which is the preferred way because it's more flexible). I personally think that we should keep this convention here to make it very clear if the result is stack allocated or heap allocated, as otherwise we could run into seg-faults and double-frees because we forget if a `deinit` function frees the pointer or not.

Reply:
That makes sense :)